lib/merge.js

```diff
@@ -9,21 +9,39 @@ const assert = require('assert');
  * @ignore
  * @memberof cloudfriend
  * @name merge
- * @param {...object} template - a CloudFormation template to merge with
+ * @param {...object} templates - a CloudFormation template to merge with
  * @returns {object} a CloudFormation template including all the Metadata,
  * Parameters, Mappings, Conditions, Resources, and Outputs from the input
  * templates
  * @throws errors when there is overlap in logical resource names between
  * templates
  */
-module.exports = function() {
+module.exports = function(...templates) {
   const template = {
     AWSTemplateFormatVersion: '2010-09-09',
+    /**
```

Comment:
These `@type {object}` are needed to allow for subsequent addition of more properties on each of these objects. Otherwise `tsc` treats each one as an independent type that should not have keys added/removed.